pkg/batcher/batcher.go

```diff
@@ -118,26 +130,39 @@ func (b *Batcher[T, U]) run() {
 		select {
 		// context that we started with has completed so the app is shutting down
 		case <-b.ctx.Done():
+			_ = b.requestWorkers.Wait()
 			return
 		case <-b.trigger:
 			// wait to start the batch of create fleet calls
 		}
 		b.waitForIdle()
-		b.runCalls()
+
+		// Copy the requests, so we can reset the requests for the next batching loop
+		b.mu.Lock()
+		requests := b.requests
+		b.requests = map[uint64][]*request[T, U]{}
+		b.mu.Unlock()
+
+		for _, v := range requests {
+			req := v // create a local closure for the requests value
+			b.requestWorkers.Go(func() error {
+				b.runCalls(req)
+				return nil
+			})
+		}
 	}
 }
 
 func (b *Batcher[T, U]) waitForIdle() {
 	timeout := time.NewTimer(b.options.MaxTimeout)
 	idle := time.NewTimer(b.options.IdleTimeout)
-	count := 0
-	for {
+	count := 1 // we already got a single trigger
+	for b.options.MaxItems == 0 || count < b.options.MaxItems {
```

Comment:
This should be `<=` right? 

```suggestion
	for b.options.MaxItems == 0 || count <= b.options.MaxItems {
```

Reply:
If it's less than or equal to, then we would always be over by 1. We would continue looping when count == b.options.MaxItems and then would only exit after we get triggered one more time